tests_integration/snowflake_connector.py

```diff
@@ -10,46 +10,34 @@
 _ENV_PARAMETER_PREFIX = "SNOWFLAKE_CONNECTIONS_INTEGRATION"
 
 
-@pytest.fixture(scope="session")
-def create_database():
-    database_name = f"db_{uuid.uuid4().hex}"
+@pytest.fixture(scope="function")
+def database_name():
+    yield f"db_{uuid.uuid4().hex}"
 
+
+@pytest.fixture(scope="function")
+def snowflake_session(database_name):
```

Comment:
Is it possible to have two fixtures like following:

* `test_database` - scope "function" - responsible for creating a database, setting "use database" and "use schema", setting env variables, dropping the database and removing the environment variables
* `snowflake_session` - scope "session" - responsible for creating a connection to snowflake before all tests and closing it after all tests. The connection would be configured without database level settings (database, schema).

IMO it could allow to have much less connections. WDYT? Is it possible?

Reply:
Good idea, but it requires to add `database_name` to all tests

Reply:
Nvm it is a great idea :)